### PR TITLE
Enforce prettier

### PR DESCRIPTION
**Pull request by kartojal**
- Add Lint step at CI to prevent to merge unformatted code into the repository
- Run `prettier -w .` to format the entire codebase

Tried with different values of printWidth to minimize the diff impact of this PR, but as expected there is even greater diff (over 6k lines diff) if you move from the default 100 char of printWidth.

### CODE REVIEW

File: .husky/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_
```
Comment (miguelmtzinf) [inline .husky/.gitignore:1]: is inteded to be empty?

File: .prettierignore [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
 node_modules
+types
+deployments
+temp-artifacts
+dist
+Certora
+*.json
+CHANGELOG.md
+.prettierignore
+.solcover.js
```
Comment (miguelmtzinf) [inline .prettierignore:12]: should we included `.solcover.js`?
and `Certora`? At the end of the day, their source code should be the same the `contracts` one

File: .husky/.gitignore (hunk 1 above)
Comment (kartojal) [reply .husky/.gitignore:1]: is added by husky, not sure about the meaning

File: .prettierignore (hunk 2 above)
Comment (kartojal) [reply .prettierignore:12]: added to ignore to prevent more diffs but I understand your point

File: contracts/interfaces/IPoolDataProvider.sol [hunk 3]
```diff
@@ -222,10 +230,9 @@ interface IPoolDataProvider {
    * @param asset The address of the underlying asset of the reserve
    * @return irStrategyAddress The address of the Interest Rate strategy
    */
-  function getInterestRateStrategyAddress(address asset)
-    external
-    view
-    returns (address irStrategyAddress);
+  function getInterestRateStrategyAddress(
```
Comment (ShayneMichaelBacik123) [inline contracts/interfaces/IPoolDataProvider.sol:233]: Well i would approve of this. However it seems it was developed by anger or just plain selfishness but I could be wrong I'll take a chance.

**Review by ShayneMichaelBacik123 — COMMENTED** [review]
Thanks

**Comment by miguelmtzinf** [comment]
Closes #788

**Review by miguelmtzinf — APPROVED** [review]

**Review by foodaka — APPROVED** [review]

**Pull request merged**